### PR TITLE
adds to do list

### PR DESCRIPTION
**Pull request by visha12l**

### CODE REVIEW

File: src/javascript/todoList/todoItem.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    render() {
+        let listData = this.props.listData;
+        return (
+            <div>
+                <ul className="todoList">
+                    {underscore.map(listData, (list, key) => {
+                        return (
+                            <li key={key} className="clearFloat">
+                                <p className="floatLeft">Task::{list.userInput}</p>
+                                <a className="btn redButton floatRight" onClick={this.deleteItem.bind(this, key)}>delete</a>
```
Comment (swapnil-webonise) [inline src/javascript/todoList/todoItem.js]: Bind such methods in constructor

File: src/javascript/todoList/todoList.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import TodoItem from './todoItem.js';
+import TodoSearch from './todoSearch.js';
+import '../../stylesheets/todolist.css';
```
Comment (swapnil-webonise) [inline src/javascript/todoList/todoList.js]: Here you can use path alias which we can configure in webpack like

    resolve: {
        root: path.join(APP, 'assets'),
        alias: {
            stylesheets: '../../stylesheets'
        }
    }

File: src/javascript/todoList/todoList.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import React from 'react';
```
Comment (swapnil-webonise) [inline src/javascript/todoList/todoList.js]: You can mention commonly used packages in web pack provide plugin like

```
plugins: [
      new webpack.ProvidePlugin({      
             React: 'react',      
             PropTypes: 'prop-types'    
      }),
]
```

File: src/javascript/todoList/todoList.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        let userInput= this.refs[refName].value;
+        if(userInput) {
+            this.setState({
+                addbuttonStatus: true,
+            });
+        } else {
+            this.setState({
+                addbuttonStatus: false,
+            });
+        }
```
Comment (swapnil-webonise) [inline src/javascript/todoList/todoList.js]: You can write it like this

```
 this.setState({
       addbuttonStatus: !!userInput,
 });
```

File: src/javascript/todoList/todoItem.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+var underscore = require('underscore');
+
+export default class TodoItem extends React.Component {
```
Comment (swapnil-webonise) [inline src/javascript/todoList/todoItem.js]: Always gives default props to a component

File: src/javascript/todoList/todoList.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        return true;
+    }
+
+    addItem(event) {
+        event.preventDefault();
+        let noteText = this.refs.input.value.toLowerCase() // user input value
+        let result = this.state.oldList; //getting old list
+        let duplicateEntry = underscore.contains(underscore.pluck(result, 'userInput') , noteText); //check for duplicate value
+        if(noteText && !duplicateEntry) {
+            result.push({userInput:noteText});
```
Comment (swapnil-webonise) [inline src/javascript/todoList/todoList.js]: NEVER mutate this.state directly, You use concat method for such things, it creates a copy with new data

```
var joined = this.state.oldList.concat({userInput:noteText});
this.setState({ oldList: joined })
```

File: src/javascript/todoList/todoList.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        } else {
+            this.setState({
+                requserrorMsg: true,
+            });
+        }
+          this.refs.input.value="";
+    }
+
+    deleteItem (deleteItemId) { //delete item id
+        let result = this.state.oldList;
```
Comment (swapnil-webonise) [inline src/javascript/todoList/todoList.js]: Same here on state mutation

File: src/javascript/todoList/todoList.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+                    <h3>{oldList.length} task to show </h3>
+                    <div className="clearFloat">
+                        <div className="floatLeft">
+                            <div className="clearFloat">
+                                <div className="userData floatLeft">
+                                    <input placeholder="Enter Item"
+                                        ref="input"
+                                        onChange={this.handleInputChange.bind(this,"input")}
+                                        onFocus={this.handleInputFocus.bind(this, "floatingLabel")}
+                                        onBlur={this.handleInputBlur.bind(this, "floatingLabel")}
```
Comment (swapnil-webonise) [inline src/javascript/todoList/todoList.js]: Same here bind such functions in constructor

File: src/javascript/todoList/todoItem.js (hunk 1 above)
Comment (visha12l) [reply src/javascript/todoList/todoItem.js]: done

File: src/javascript/todoList/todoList.js (hunk 4 above)
Comment (visha12l) [reply src/javascript/todoList/todoList.js]: done

File: src/javascript/todoList/todoList.js (hunk 6 above)
Comment (visha12l) [reply src/javascript/todoList/todoList.js]: state mutation removed

File: src/javascript/todoList/todoList.js (hunk 7 above)
Comment (visha12l) [reply src/javascript/todoList/todoList.js]: state mutation removed

File: src/javascript/todoList/todoList.js (hunk 8 above)
Comment (visha12l) [reply src/javascript/todoList/todoList.js]: done

File: src/javascript/todoList/todoItem.js (hunk 5 above)
Comment (visha12l) [reply src/javascript/todoList/todoItem.js]: added default props to the component

File: src/javascript/todoList/todoItem.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+var underscore = require('underscore');
```
[the remark this answers is not in the document]
Comment (visha12l) [reply src/javascript/todoList/todoItem.js]: done

**Pull request merged**